src/platform/ESP32/BLEManagerImpl.h

```diff
@@ -25,6 +25,7 @@
 
 #pragma once
 #include <sdkconfig.h>
+#include <string>
```

Comment:
https://github.com/project-chip/connectedhomeip/pull/38067/files#diff-2f4f687a9c0b40262a3578507f24278aedaf5b9e3a00c78b2e479db7e4ac49b3R285 adds a vector usage ... why do we include string?

Reply:
we seem to need vector if CONFIG_BT_NIMBLE_ENABLED

```suggestion
#include <string>

#if CONFIG_BT_NIMBLE_ENABLED
#include <vector>
#endif
```

Also line 283 uses `defined(...)` ... can/should we change that to a plain if?

Reply:
Done. The defined(...) is introduced as a part of [this](https://github.com/project-chip/connectedhomeip/pull/29728) PR for -Wundef build errors.